components/ProposalVotingPower/NftVotingPower.tsx

```diff
@@ -36,50 +27,19 @@ const Join = () => {
   const wallet = useWalletOnePointOh()
   const connected = !!wallet?.connected
   const realm = useRealmQuery().data?.result
+  const { userNeedsTokenOwnerRecord, handleRegister } = useJoinRealm();
 
-  const ownTokenRecord = useUserCommunityTokenOwnerRecord().data?.result
-  const client = useVotePluginsClientStore(
-    (s) => s.state.currentRealmVotingClient
-  )
-
-  const handleRegister = async () => {
-    if (!realm || !wallet?.publicKey || !client.client) throw new Error()
+  // TODO [CT] as per GatewayCard, we might want a separate join button that is not specific to the nft plugin
+  // this is no longer any way related to the nft plugin
+  const join = async () => {
```

Comment:
❤️

Reply:
This is resolved in the third milestone PR. I'll remove the TODO here for neatness.